### PR TITLE
H-3304: Initial setup for `SqlFluff`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We don't have an SQL formatter and SqlFluff seems the best suited one out there. This adds a basic setup to the CI and a simple configuration. 

## ❓ How to test this?

- Install the linter using `brew install sqlfluff`
- Run it with `sqlfluff fix`

### Adding it to Zed as autoformatter:
Merge this snippet into your configuration:
```json

  "languages": {
    "SQL": {
      "formatter": [
        {
          "external": {
            "command": "sqlfluff",
            "arguments": ["fix", "-"]
          }
        }
      ]
    }
  }
```

### CODE REVIEW

File: .sqlfluff [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[sqlfluff]
+dialect = postgres
+encoding = utf-8
+exclude = ["libs/@local/graph/postgres-store/postgres_migrations/"]
```
Comment (TimDiekmann) [inline .sqlfluff:4]:
```suggestion
```

File: libs/@local/graph/migrations/graph-migrations/v009__entities/up.sql [hunk 2]
```diff
@@ -77,7 +77,7 @@ CREATE TABLE entity_has_left_entity (
     FOREIGN KEY (left_web_id, left_entity_uuid) REFERENCES entity_ids
 );
 CREATE INDEX entity_has_left_entity_source_idx
-    ON entity_has_left_entity(web_id, entity_uuid);
```
Comment (indietyp) [inline libs/@local/graph/migrations/graph-migrations/v009__entities/up.sql:80]: I am not sure if I am a huge fan of the removal of indentation here (or citus one), but no strong opinions really.

**Review by indietyp — APPROVED** [review]
Thank you!

File: libs/@local/graph/migrations/graph-migrations/v009__entities/up.sql (hunk 2 above)
Comment (TimDiekmann) [reply libs/@local/graph/migrations/graph-migrations/v009__entities/up.sql:80]: Probably this is configurable. I'll take a Quick Look

Comment (TimDiekmann) [reply libs/@local/graph/migrations/graph-migrations/v009__entities/up.sql:80]: Found for `JOIN` but not for creating indices/extensions. Also not a strong opinion, so I'll just merge it 🙂

**Pull request merged**